pcsx2/GS/GSCapture.cpp

```diff
@@ -545,7 +545,7 @@ bool GSCapture::BeginCapture(float fps, GSVector2i recommendedResolution, float
 			return false;
 		}
 
-		s_converted_video_frame->format = sw_pix_fmt;
+		s_converted_video_frame->format = s_video_codec_context->pix_fmt;
```

Comment:
Are you sure about this one? For hardware encoders, the context pixel format may be different, and we want to convert from RGB->YUV (or whatever sw_pix_fmt is), then to hw.

Reply:
I'm not sure about this one, none of the macOS encoders return something that uses an AVCodecHWConfig

Things I can confirm:
- The macOS hardware encoder, h264_videotoolbox, does not have a AVCodecHWConfig, but offers the pixel formats `AV_PIX_FMT_VIDEOTOOLBOX` (hw pixel format), `AV_PIX_FMT_NV12`, and `AV_PIX_FMT_YUV420P`.  If we request `AV_PIX_FMT_YUV420P`, the resulting format on `s_video_codec_context` is `AV_PIX_FMT_YUV420P`.
- If you select FFV1 and add `pixel_format=yuv444p` to the extra arguments, you get broken output if you try to feed it yuv420p frames.

Reply:
Ahh I see where we write the hw format in
I changed it to only use that one if the user specified a `pixel_format` argument